src/internals/ir/callGraph.ts

```diff
@@ -0,0 +1,193 @@
+import { TactASTStore } from "./astStore";
+import { IdxGenerator } from "./indices";
+import {
+  AstFunctionDef,
+  AstReceiver,
+  AstContractInit,
+  AstStatement,
+  AstStaticCall,
+  AstMethodCall,
+  AstExpression,
+  AstStructFieldInitializer,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+class CGEdge {
+  public idx: number;
+  constructor(
+    public src: number,
+    public dst: number,
+  ) {
+    this.idx = IdxGenerator.next("edge");
+  }
+}
+
+class CGNode {
+  public idx: number;
+  constructor(
+    public astId: number,
+    public name: string,
+  ) {
+    this.idx = IdxGenerator.next("node");
+  }
+}
+
+export class CallGraph {
+  public nodes: CGNode[] = [];
+  public edges: CGEdge[] = [];
+  private nodeMap: Map<number, CGNode> = new Map();
+  private edgeMap: Map<number, Set<number>> = new Map();
+
+  constructor(public astStore: TactASTStore) {}
```

Comment:
```suggestion
  constructor(public readonly astStore: TactASTStore) {}
```

Or even better: move it as an argument of the `build` function.